guardian/prosemirror-typerighter
# Offset API for 'scroll to rule' behaviour

## What does this change?

This PR adds an offset API for the 'scroll to rule' sidebar behaviour. The example implementation centres the rules in the editor.

![scroll-to-match](https://user-images.githubusercontent.com/7767575/90252585-a883d100-de37-11ea-9de4-f3aa224fbef5.gif)

## Implementation notes

Because we don't know much about our consumer's editor, we defer to them with a `getScrollOffset` callback.

It's a callback, rather than a static value, because the editor size may change during the page lifecycle (for example, if the user alters the window size), and it's likely we'd need to adjust accordingly to e.g. centre matches correctly.

## How to test

In the example implementation, click a rule in the sidebar in a long-ish document. The editor should scroll to the match in the document, centering it on the page.

## How can we measure success?

Plugin consumers have more control about how the plugin scrolls matches into view.
